### DISCUSSION TITLE
Getting Started with Seeed Studio XIAO ESP32S3 (Sense) | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#124] [Q&A]
# Getting Started with Seeed Studio XIAO ESP32S3 (Sense) | Seeed Studio Wiki

Getting Started with Seeed Studio XIAO ESP32S3.

https://wiki.seeedstudio.com/xiao_esp32s3_getting_started/

**Comment by ItsJohnnyB** [comment]
After adding the XIAO_ESP32S3 board and variant files it help to restart the Arduino app to refresh the list of boards to select.  --  Johnny B

**Comment by Lenbus** [comment]
Really love this thing. To integrate with my workflow I really need to get this to work with Platform.IO. 
Any hints how to do that as the board manager of PIO does not seem to support this board yet? :)

**Reply by matsujirushi** [reply]
Hi,
How about specifying the latest [arduino-esp32](https://github.com/espressif/arduino-esp32) in [platform_packages](https://docs.platformio.org/en/stable/projectconf/sections/env/options/platform/platform_packages.html)?

**Reply by Love4yzp** [reply]
The platform requires a file called `seeed_xiao_esp32s3.json'.
However, according to the wiki, by replacing some files with platformio, you can use the intelligence plugin.

**Comment by Love4yzp** [comment]
Why it's unable to use `Serial.println()`?
I didn't get any info available from XIAO_ESP32s3 when working with serial console of Arduino IDE.

**Reply by Love4yzp** [reply]
OK, the problem is solved: Enable the USB CDC option.

**Reply by Gogny** [reply]
On PlatformIO you can use these build flags :
```
build_flags = 
	-DARDUINO_USB_MODE=1
	-DARDUINO_USB_CDC_ON_BOOT=1
```

**Comment by FedericoBusero** [comment]
In Arduino ESP32 board version 2.0.7 is buggy: it can corrupt esptool so that you can no longer any upload code to any ESP32 chip (see https://github.com/espressif/arduino-esp32/issues/7923)

The good news
1) It is fixed in board version 2.0.8
2) Version 2.0.8 includes the XIAO_ESP32S3 board, so the manual step about boards.txt and variant subfolder is no longer necessary (tested on Windows)

**Comment by FedericoBusero** [comment]
Cool board! Porting Arduino code from ESP32CAM was rather easy. The Arduino example code of CameraWebServer shows the correct pin definitions.

Some things I learned while porting my motor/servo controller + video streamer:
- the naming of the pins (D1, D2, ...) is not the same as the GPIO numbering, so you need to use the D-number in Arduino code (just like ESP8266 chips). Rather uncommon for a ESP32 chip.
- the ESP32S3 chip has 4 ledc Timers, but each timer only has 2 channels using the ledc API. Channel allocation (led, motor, servo) should be checked. I dropped the PWM of the builtin led as the led of XIAO is of course weak compared to the flash led of ESP32CAM. Also the bit-width of ledc is only 14 bit, so I had to adapt the servo calculation a bit. With 1 timer used by the camera, you can still use 3 groups of 2 servo's/motors using the standard ledc api.

I developed a motor & servo controller & "video" stream of the camera from a Blimp to the browser of a smartphone, with one joystick and one slider in the web UI: https://github.com/FedericoBusero/ESP32-CAM_ZeppelinCAM

**Reply by MatthewJeffson** [reply]
Wow, amazing!

**Comment by matsujirushi** [comment]
**Seeed Studio XIAO ESP32S3 Eagle footprint** does not have JTAG and BAT, USB, Thermal pads.
Could you please add it?

![image](https://github.com/Seeed-Studio/wiki-documents/assets/8079404/7f3ff94a-11bf-46ff-ba39-8fe6635284a5)

**Reply by MatthewJeffson** [reply]
Hi, the eagle footprint offers the dimension of the board to the user should be fine since it doesn't require additional silk screen.

**Reply by matsujirushi** [reply]
**Seeed Studio XIAO ESP32S3 Eagle footprint** seems to have no following pads. (red rectangle)

![image](https://github.com/Seeed-Studio/wiki-documents/assets/8079404/fd1c8251-f047-4db2-8a85-55943948a74f)

**Reply by MatthewJeffson** [reply]
Yeah, when people use the eagle footprint for designing their own board. They usually just need the dimension of the board and don't require other things, like the red rectangle you mentioned. I got it from the electronics engineer and the product manager.

**Reply by Gogny** [reply]
Could Seeeduino update their **KICAD** footprint for the xiao ESP32S3 and ESP32C3 ? I really need the bottom pads, especially the battery pins, and I am sure I am not the only one who needs access to these.
The files of the S3 are copy-pasted from the C3 so they have the same file names, could they correct that too ?

**Reply by MatthewJeffson** [reply]
Hello, I think that about Seeeduino XIAO we have published the KiCAD files, refer to [here](https://wiki.seeedstudio.com/Seeeduino-XIAO#resourses).
And can I ask what do you mean by "The files of the S3 are copy-pasted from the C3 so they have the same file names"?

**Reply by Gogny** [reply]
Hello sir,
I downloaded the ` [ZIP] Seeed Studio XIAO ESP32S3 Sense KiCAD Libraries ` from this page and the pads are not present for all of the connexions in red as shown on the picture posted by matsujirushi. We need them for PCB design. But there is only pads for connexions on the side of the Xiao. We need the footprint for battery connexions on the bottom of the Xiao.  
I followed the link you gave me but sadly these pads are not present on the SAMD21 version, only on the ESP32 versions.   
We really need them, could you please add them to the ` .kicad_mod ` file ? This will be very useful for other clients too...
  
About the naming of the files, if you download them you will see a folder like this :  
```
- Seeeduino-xiao-ESP32S3-KiCAD-Library  
  |- Seeeduino xiao ESP32C3 KiCAD Library  
    |- xiao ESP32C3_PCB.pretty
    |- MOUDLE-SEEEDUINO-XIAO-ESP32C3.kicad_sym
```
and the schematic editor the Xiao is also labeled ` MOUDLE-SEEEDUINO-XIAO-ESP32C3 ` but it is the files for the S3 board not the C3. Is it normal ?

**Reply by MatthewJeffson** [reply]
Hi, I am terribly sorry for such mistake. I will double check with upload files. 
And I will also check with the footprint for battery connexions on the bottom of the Xiao, and upload it then if there doesn't exists.

**Reply by theepicflyer** [reply]
Yes, would like the KiCAD and EAGLE footprints to include the pads on the bottom as well. There are also some inconsistencies between the images, the libraries and the DXF file as well. For example, the DXF does not include the thermal pad. 

Please check that everything is correct and update all the files and images!

**Reply by MatthewJeffson** [reply]
We do have the footprints. And regard the files I have asked my college to change it! Thank you!

![](https://files.seeedstudio.com/wiki/SeeedStudio-XIAO-ESP32S3/img/3.png)

![](https://files.seeedstudio.com/wiki/SeeedStudio-XIAO-ESP32S3/img/2.jpg)

**Reply by madnordski** [reply]
My XIAO ESP32-S3 board does not contain reset or boot buttons. The sales page diagram and in the diagram in this post thread say it should. So far it's been loading code without complaint but I've used ESP's long enough to know that could change. Did I just get a defective board or is the documentation in error because it's trying to cover the sense board in addition to this one?  Finally, there are pads where the buttons are supposed to be -- does shorting the pads to ground work in place of the buttons?

**Reply by valerioc-unibo** [reply]
Still no bottom pads for the KiCAD footprint and symbol. Any news? Also an updated .dxf for the ESP32S3 (not showing the thermal pad) can be useful.

**Comment by matsujirushi** [comment]
XIAO-ESP32S3-Sense-housing-design(top).stp cannot be downloaded. 😢

![image](https://github.com/Seeed-Studio/wiki-documents/assets/8079404/3ceae16d-7755-46ec-a54b-7e2fd3a23f21)

**Reply by MatthewJeffson** [reply]
Hi, I will check with that! 
Thank you!

**Comment by theepicflyer** [comment]
Hi there. Is CircuitPython support being worked on? It would be very useful for me and others in the community too.

**Reply by qutefox** [reply]
It seems MicroPython is a possibility: https://forum.seeedstudio.com/t/seed-xiao-esp32-s3/270594
But for some reason the wiki doesn't talk about it. A guide would be appreciated!

**Reply by tracteurblinde** [reply]
The Generic ESP32-S3 (SPIRAM Octal) image worked for me. https://micropython.org/download/GENERIC_S3_SPIRAM_OCT/
I flashed it successfully using both the uf2 instructions above and directly using esptool.

**Comment by Maurice1995** [comment]
I have added the Seeed Studio XIAO ESP32S3 and the Seeed Studio XIAO ESP32S3 to Grabcad. If you are interested in the .stp files feel free to download and use them to your preference. If you have any suggestions for correction or improvement, please let me know on Grabcad in the comments.

https://grabcad.com/library/seeed-studio-xiao-esp32s3-1

https://grabcad.com/library/seeed-studio-xiao-esp32s3-sense-1

**Reply by MatthewJeffson** [reply]
Wow, it is very kind of you!

**Comment by FedericoBusero** [comment]
In the Arduino IDE, PSRAM is disabled by default. Wouldn't it be better to enable it by default?

**Comment by notevanrutten** [comment]
Does this board support the auto-focus capability of the OV5640 camera?

**Reply by MatthewJeffson** [reply]
yes, actually you can find the specific OV5640 camera for XIAO ESP32S3 Sense [here](https://www.seeedstudio.com/OV5640-Camera-for-XIAO-ESP32S3-Sense-With-Heat-Sink-p-5739.html)

**Comment by pat-2010** [comment]
XIAO_ESP32S3, ESP32C3 and ESP32S3 sense  all show board info of:
DFRobot Beetle ESP32-C3
VID: 0x303A
PID 0x1001
SN: (null)

using Arduino IDE 2.1.1, and rev 2.0.9 of the esp32 board manager. 

Probably related, if I start the Arduino IDE with the XIAO installed, I get an error message of "Error communication with the language server. Error: write EOF: undefined. 

I can then change the board type to XIAO_ESP32S3 (or C3)  and upload ok, but if I open a new sketch or reset the board, the board type associated with that com port changes back to the DFRobot Beetle. On the C3 that includes when I reset the XIAO_ESP32C3

**Reply by pat-2010** [reply]
I asked the same question in the seeedstudio forum. (https://forum.seeedstudio.com/t/xiao-esp32-identified-by-arduino-ide-as-dfrobot-beetle/271328/3). 

The response below pasted from that forum answers the question.

[Seeed_Jostar](https://forum.seeedstudio.com/u/Seeed_Jostar)

Yes, XIAOesp32c3 is not officially released by Arduino, so XIAOesp32 cannot be correctly recognized by arduino ide, so it needs to be manually set to the correct development board (XIAO-ESP32C3) though the arduino ide recognizes it as another board  😂

**Comment by MARXTECHDIY** [comment]
I just got my XIAO esp32s3 sense  and my IDE keeps telling me it's a WIFIduinoV2. It also sees it as XIAO_ESP32S3 when I search for it. Everything works fine, but then it always gives me:
Access is denied.
Exit Status 1.

Does anyone know how I can fix this?

**Comment by jgrissom** [comment]
Is it safe to connect a 3.7 volt lipo battery to the 3.3v gpio and ground gpio pins for power? It seems to work fine.

**Comment by hpsaturn** [comment]
I had a bad quality problem with this module. If someone maybe have the next error:

```bash
E (5181) camera: Camera probe failed with error 0x105(ESP_ERR_NOT_FOUND)
```
maybe is its sensor module has a physical problem or damage.

I explained it in detail in the next repository, because I didn't found the official source code or repository from Seedstudio for this camera and its examples:

https://github.com/limengdu/SeeedStudio-XIAO-ESP32S3-Sense-camera/issues/8

**Comment by scottw-finao** [comment]
what is the pin number for the builtin led?

**Reply by pavloshargan** [reply]
21

**Reply by pavloshargan** [reply]
And keep in ming that the LED is on when It's set to LOW

**Comment by Muditha-Kumara** [comment]
What is cutoff battery voltage? Is this included a bulk converter?

**Comment by tsujido3r** [comment]
How can i connect a inductive wireless charger to this module for charging the battery. I assume its just a direct conncetion to the VBUS pin.

**Comment by Focus06** [comment]
Hello,
[I have this one](https://www.tme.eu/pl/details/seeed-113991115/zestawy-do-transmisji-danych/seeed-studio/xiao-esp32s3-sense/). And ESP 32 CAM borrowed. I did everything from [this book](https://mjrovai.github.io/XIAO_Big_Power_Small_Board-ebook/chapter_1-1.html) and a lot more such installing drivers CH34x and cp21xx. I'm still having an issue with flashing the program. Buy those previous actions I shifted from "exit status 2" to "exit status 1" (Arduino IDE 2).  I seems my PC sees device now, Arduino sees the port but can not perform flashing. At the same time Arduino UNO works fine and flashes. I did similar or the same actions on 6 PCs including Windows 8, 10, 11, Ubuntu... Some of them with support of team mate. Please give me advices what went wrong? Why original UNO works and SEEEDUINO with proper plugins not?

**Reply by feiticeir0** [reply]
Before connecting the Xiao to your computer,  you need to have the boot button pressed, connect the USB cable and then you can release the button. That way you can upload the sketch

**Reply by Focus06** [reply]
Thanks feiticeir0 for quick response! There are few things I discovered:
1) On UBUNTU I had to switch on boards: ESP32 by Espressif Systems -> XIAO_ESP32S3 -- it works
a) Now I'm using more "professional" tool to hold "boot" button - maybe I had in past some debouncing artefact caused by "weak" connection when releasing
b) After flashing I'm clicking "reset" button 
c) basic blinking code works
2) On Windows 11 still having issue. Sequence as above is not working.

**Comment by autume** [comment]
Hi

**Comment by valnenov** [comment]
I am tying to load a sketch on  Xiao ESP32-S3 using Arduino IDE running on an iMac. The IDE does not see the  port. I have tried all suggestions that I can come across including holding the boot button while connecting the USB cable to the Mac but no luck. 

Here is the error I get: Can any one please help?

Arduino: 1.8.19 (Mac OS X), Board: "XIAO_ESP32S3, Disabled, Disabled, QIO 80MHz, 8MB (64Mb), Core 1, Core 1, Hardware CDC and JTAG, Disabled, Disabled, Disabled, UART0 / Hardware CDC, Default with spiffs (3MB APP/1.5MB SPIFFS), 240MHz (WiFi), 115200, None, Enabled"

Sketch uses 267417 bytes (8%) of program storage space. Maximum is 3342336 bytes.
Global variables use 17888 bytes (5%) of dynamic memory, leaving 309792 bytes for local variables. Maximum is 327680 bytes.
/Users/blabla/Library/Arduino15/packages/esp32/tools/esptool_py/4.6/esptool --chip esp32s3 --port /dev/cu.usbserial-1410 --baud 115200 --before default_reset --after hard_reset write_flash -e -z --flash_mode keep --flash_freq keep --flash_size keep 0x0 /var/folders/dg/c3ggmmbs3c50nh_8fbjlss480000gn/T/arduino_build_941635/buzzer.ino.bootloader.bin 0x8000 /var/folders/dg/c3ggmmbs3c50nh_8fbjlss480000gn/T/arduino_build_941635/buzzer.ino.partitions.bin 0xe000 /Users/blabala/Library/Arduino15/packages/esp32/hardware/esp32/3.0.5/tools/partitions/boot_app0.bin 0x10000 /var/folders/dg/c3ggmmbs3c50nh_8fbjlss480000gn/T/arduino_build_941635/buzzer.ino.bin 
esptool.py v4.6
An error occurred while uploading the sketch
Serial port /dev/cu.usbserial-1410

A fatal error occurred: Could not open /dev/cu.usbserial-1410, the port doesn't exist

**Reply by Focus06** [reply]
Hi valnenov, try this sequence
1) connect USB, pick port in Arduino IDE
2) Push and hold BOOT (up to point 4)
3) Click RESET
4) Wait around 1 second and release BOOT
Use tools that guarantee good contact for those thiny buttons, try not to let them bounce.
If didn't work at first time try few (2-4). In my case it finaly flashes to ESP. I had this kind of issue but works now when I use this order.

**Comment by pavloshargan** [comment]
Here is a simple example that lights a built-in led (yellow one) when the built in button ( boot ) is pressed. May be useful for the start:

```
// constants won't change. They're used here to set pin numbers:
const int buttonPin = 0;  // the number of the pushbutton pin
const int ledPin = 21;    // the number of the LED pin

// variables will change:
int buttonState = 0;  // variable for reading the pushbutton status

void setup() {
  // initialize the LED pin as an output:
  pinMode(ledPin, OUTPUT);
  // initialize the pushbutton pin as an input:
  pinMode(buttonPin, INPUT);
}

void loop() {
  // read the state of the pushbutton value:
  buttonState = digitalRead(buttonPin);

  // check if the pushbutton is pressed. If it is, the buttonState is HIGH:
  if (buttonState == LOW) {
    // turn LED on:
    digitalWrite(ledPin, LOW);
  } else {
    // turn LED off:
    digitalWrite(ledPin, HIGH);
  }
}
```

**Comment by gemini86** [comment]
Just got started with these boards for their dual cores. I'm using arduino IDE. First I updated all boards and libraries, all currently up to date. Then I find my board on COM8, verify and then compile/upload. This fails after a minute with "A fatal error occurred: Could not open COM8, the port doesn't exist" but the device is still there and listen. I followed the troubleshooting guide, put it into bootloader mode (hold boot while plugging in the usb cable) and this still does nothing, but I also think it's dropping out of bootloader mode before I actually get to the upload portion. I eventually tried using the UF2 converter tool in the guide, so I exported my .bin and converted, put it into bootloader mode, drop the uf2 file in and it runs. I can communicate with my firmware over the arduino serial monitor. BUT, my sketch uses OneWire and DallasTemperature to read a ds18b20 probe, which is not working. After touching the reset button a couple times, it seems my firmware is no longer running, so I go into bootloader mode and re-upload mu uf2 file, try again, and it's working. This process repeats for some time. Anyone know if this board is unable to work with the OneWire or DallasTemperature libraries?

**Comment by KR0707** [comment]
ESP32S3's library for eagle is now for ESP32C3.
Will you fix it?

**Reply by KR0707** [reply]
Sry, my mistake.
I found it.

**Comment by gelit** [comment]
I try a source code **to use ST7735 with XIAO ESP32S3 Sense**
Thanks in advance
